test/Gate.js

```diff
@@ -0,0 +1,294 @@
+// Copyright 2017 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Test: Gate.js
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const Utils = require('./lib/utils.js');
+const BigNumber = require('bignumber.js');
+const Gate_utils = require('./Gate_utils.js');
+const HashLock = require('./lib/hash_lock.js');
+
+contract('Gate', function(accounts) {
+
+  var stakerAccount = accounts[0]
+    , stakeAmount = new BigNumber(web3.toWei(1000, "ether"))
+    , beneficiaryAccount = accounts[6]
+    , workerAddress1 = accounts[7];
+  ;
+
+  const deployGate = async function() {
+    result   = await Gate_utils.deployGate(artifacts, accounts);
+    valueToken  = result.valueToken;
+    openSTValue = result.openSTValue;
+    uuid = result.uuid;
+    gate = result.gate;
+    workers = result.workers;
+    bounty = result.bounty;
+  };
+
+  const requestStake = async function () {
+    let initialStakerAccountBalance = await valueToken.balanceOf.call(stakerAccount)
+      , initialGateBalance = await valueToken.balanceOf.call(gate.address)
+    ;
+    // approve gate contract
+    await valueToken.approve(gate.address, stakeAmount, { from: stakerAccount });
+
+    let requestStakeResult = await gate.requestStake.call(stakeAmount, beneficiaryAccount, {from: stakerAccount});
+    assert.equal(requestStakeResult, true);
+
+    let requestStakeResponse = await gate.requestStake(stakeAmount, beneficiaryAccount, {from: stakerAccount});
+    await Gate_utils.checkRequestStakeEvent(requestStakeResponse.logs[0],stakerAccount, stakeAmount, beneficiaryAccount);
+
+    let finalStakerAccountBalance = await valueToken.balanceOf.call(stakerAccount)
+      , finalGateBalance = await valueToken.balanceOf.call(gate.address)
+    ;
+    // check balances
+    assert.equal(finalStakerAccountBalance.equals(initialStakerAccountBalance.sub(stakeAmount)), true);
+    assert.equal(finalGateBalance.equals(initialGateBalance.plus(stakeAmount)), true);
+
+    // request again should fail
+    await Utils.expectThrow(gate.requestStake(stakeAmount, beneficiaryAccount, {from: stakerAccount}));
+  };
+
+  const acceptStakeRequest = async function() {
+    await valueToken.transfer(workerAddress1, new BigNumber(web3.toWei(10000, "ether")),{from: accounts[0]});
```

Comment:
accounts[7] also has large balance for testing purposes; seems not needed?

Reply:
When I call ``` await valueToken.balanceOf.call(workerAddress1) ```, I get balance as 0. 
I checked all the account balance, only account[0] had the large balance. So I used transfer.